src/fitnesse/responders/editing/SavePropertiesResponder.java

```diff
@@ -6,38 +6,39 @@
 import fitnesse.FitNesseContext;
 import fitnesse.authentication.AlwaysSecureOperation;
 import fitnesse.authentication.SecureOperation;
-import fitnesse.authentication.SecureResponder;
 import fitnesse.http.Request;
 import fitnesse.http.Response;
 import fitnesse.http.SimpleResponse;
-import fitnesse.responders.NotFoundResponder;
+import fitnesse.responders.BasicResponder;
 import fitnesse.wiki.*;
 
 import java.util.Arrays;
 import java.util.LinkedList;
 import java.util.List;
 
-public class SavePropertiesResponder implements SecureResponder {
+public class SavePropertiesResponder extends BasicResponder {
   @Override
   public Response makeResponse(FitNesseContext context, Request request) throws Exception {
-    SimpleResponse response = new SimpleResponse();
-    String resource = request.getResource();
-    WikiPagePath path = PathParser.parse(resource);
-    WikiPage page = context.getRootPage().getPageCrawler().getPage(path);
-    if (page == null)
-      return new NotFoundResponder().makeResponse(context, request);
+    WikiPage page = getPage(context, request);
+
     PageData data = page.getData();
     saveAttributes(request, data);
     VersionInfo commitRecord = page.commit(data);
+    SimpleResponse response = new SimpleResponse();
     if (commitRecord != null) {
       response.addHeader("Current-Version", commitRecord.getName());
     }
     context.recentChanges.updateRecentChanges(page);
-    response.redirect(context.contextRoot, resource);
+    response.redirect(context.contextRoot, request.getResource());
 
     return response;
   }
 
+  @Override
+  protected String contentFrom(FitNesseContext context, Request request, WikiPage requestedPage) {
```

Comment:
Where/when is this method called (as you overridden `BasicResponder. makeResponse()`)

Maybe this implementation can be moved to BasicResponder? I believe it is also present in `DisabledResponder`, `PageDataWikiPageResponder` and `DefaultResponder`

Reply:
I'm working now on another computer. Primarily I let Intellij implement the required with method and left the implementation with `null`. Which caused a null pointer exception because some code expects to receive a `String `object but instead receiving `null`. But on this computer I don't receive a null pointer exception. I also don't find any code invoking the `contentFrom()` method. So I'm going to remove it.

Yes I will move the implementation to BasicResponder for other classes who does makes use of it 👍.